src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_header.html.twig

```diff
@@ -9,12 +9,42 @@
         </h1>
         {% include "@SyliusAdmin/Product/Show/_breadcrumb.html.twig" %}
     </div>
-    <div class="middle aligned column">
-        <div class="ui right floated buttons">
-            <a class="ui labeled icon button {% if not product.enabled %} disabled {% endif %}" href="{{ path('sylius_shop_product_show', {'slug': product.slug, '_locale': product.translation.locale}) }}">
-                <i class="angle right icon"></i>
-                {{ 'sylius.ui.show_product_in_shop_page'|trans }}
-            </a>
+    {% if product.channels|length > 1 %}
+        <div class="middle aligned column">
+            <div class="ui right floated buttons">
+                <div class="ui floating dropdown labeled icon button">
+                    <i class="share alternate icon"></i>
+                    <span class="text">
+                    {{ 'sylius.ui.show_product_in_shop_page'|trans }}
+                    </span>
+                    <div class="menu">
+                        <div class="scrolling menu">
+                            {% for channel in product.channels if channel.enabled == true %}
+                                    {% set url = channel.hostname is not null ? 'http://' ~ channel.hostname ~ path('sylius_shop_product_show', {'slug': product.slug, '_locale': channel.defaultLocale.code}) : url('sylius_shop_product_show', {'slug': product.slug, '_locale': channel.defaultLocale.code}) %}
+                                    <a href="{{ url|raw }}" class="item">
+                                        <i class="angle right icon"></i>
+                                        {{ 'sylius.ui.show_in'|trans }}
+                                        {{ channel.name }} ({{ channel.code }})
+                                    </a>
+                            {% endfor %}
+                        </div>
+                    </div>
+                </div>
+            </div>
         </div>
-    </div>
+    {% else %}
+        <div class="middle aligned column">
+            <div class="ui right floated buttons">
+                {% for channel in product.channels %}
```

Comment:
Change in the same way as you have changed foreach above

Reply:
However, I'm not sure if using foreach when you have one or zero channels is a correct solution here. What will happen if we do not have any channel?